pkg/featuregate/featuregate.go

```diff
@@ -78,6 +78,15 @@ var (
 		featuregate.WithRegisterDescription("enables features associated to the Prometheus Operator"),
 		featuregate.WithRegisterFromVersion("v0.82.0"),
 	)
+
+	// EnableTargetAllocatorRewrite is the feature gate that controls whether the collector's configuration should
+	// automatically be rewritten when the target allocator is enabled.
+	UseManifestReconciliation = featuregate.GlobalRegistry().MustRegister(
+		"operator.manifeststrategy",
+		featuregate.StageStable,
```

Comment:
Obviously I'm going to change this, but i wanted to see if all of our e2e tests would pass with this change first so I didn't need to duplicate anything